### PR TITLE
Facebook Pixel event_id discrepancy

### PR DESCRIPTION
**Pull request by rohithkaza**
## Description of the change

> Assigning event_id in device mode as per cloud mode.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

---
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/486)

### CODE REVIEW

File: integrations/FacebookPixel/browser.js [hunk 1]
```diff
@@ -310,7 +310,7 @@ class FacebookPixel {
                 : this.formatRevenue(price),
             },
             {
-              eventID: messageId,
+              event_id: traits.event_id || context.traits.event_id || properties.event_id || messageId,
```
[the remark this answers is not in the document]
Comment (rohithkaza) [reply integrations/FacebookPixel/browser.js]: Done

File: utils/getEventId.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+function getEventId(message){
+    return traits.event_id || context.traits.event_id || properties.event_id || messageId;
```
Comment (utsabc) [inline utils/getEventId.js:2]: message is passed but `traits.event_id` is referenced, but we are not extracting traits hence traits, context, properties and messageId will always be undefined

File: integrations/FacebookPixel/browser.js [hunk 3]
```diff
@@ -73,7 +74,7 @@ class FacebookPixel {
 
   page(rudderElement) {
     const { properties, messageId } = rudderElement.message;
-    window.fbq("track", "PageView", properties, { eventID: messageId });
+    window.fbq("track", "PageView", properties, { getEventId(message) });
```
Comment (akashrpo) [inline integrations/FacebookPixel/browser.js:77]: shouldn't this be `{ event_id: getEventId(message) }` ?
```suggestion
    window.fbq("track", "PageView", properties, { event_id: getEventId(message) });
```

Comment (rohithkaza) [reply integrations/FacebookPixel/browser.js]: Done

File: utils/getEventId.js (hunk 2 above)
Comment (rohithkaza) [reply utils/getEventId.js]: Committed new changes, please check once

File: utils/getEventId.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
+function getEventId(message){
+    return message.traits.event_id || message.context.traits.event_id || message.properties.event_id || message.messageId;
```
Comment (utsabc) [inline utils/getEventId.js:2]: Can you move this file to FacebookPixel folder as its specif to its use case. And rename this file as util.js

File: integrations/FacebookPixel/browser.js [hunk 5]
```diff
@@ -73,7 +74,7 @@ class FacebookPixel {
 
   page(rudderElement) {
     const { properties, messageId } = rudderElement.message;
```
[the remark this answers is not in the document]
Comment (rohithkaza) [reply integrations/FacebookPixel/browser.js]: Done

File: utils/getEventId.js (hunk 4 above)
Comment (rohithkaza) [reply utils/getEventId.js]: Done

File: integrations/FacebookPixel/browser.js [hunk 6]
```diff
@@ -199,7 +200,7 @@ class FacebookPixel {
           customProperties
         ),
         {
-          eventID: messageId,
+          event_id: getEventId(rudderElement.message),
```
Comment (saikumarrs) [inline integrations/FacebookPixel/browser.js:203]: Get the event ID once in the beginning of the function and re-use it everywhere.

Comment (rohithkaza) [reply integrations/FacebookPixel/browser.js]: Done

**Review by saikumarrs — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Pull request merged**